### ISSUE TITLE
Error when initializing repo with submodules

### ISSUE THREAD

**Issue by Azumgi** [asharov/git-hammer#20]
`python -m githammer init-project project_name ~/project_location` throws an error:
```bash
Traceback (most recent call last):
  File "/usr/local/Cellar/python/3.7.4_1/Frameworks/Python.framework/Versions/3.7/lib/python3.7/runpy.py", line 193, in _run_module_as_main
    "__main__", mod_spec)
  File "/usr/local/Cellar/python/3.7.4_1/Frameworks/Python.framework/Versions/3.7/lib/python3.7/runpy.py", line 85, in _run_code
    exec(code, run_globals)
  File "/Users/user/venv/lib/python3.7/site-packages/githammer/__main__.py", line 155, in <module>
    parsed_args.func(parsed_args)
  File "/Users/user/venv/lib/python3.7/site-packages/githammer/__main__.py", line 48, in add_repository
    hammer.add_repository(options.repository, options.configuration)
  File "/Users/user/venv/lib/python3.7/site-packages/githammer/hammer.py", line 362, in add_repository
    self._process_repository(dbrepo, session)
  File "/Users/user/venv/lib/python3.7/site-packages/githammer/hammer.py", line 295, in _process_repository
    line_counts, test_counts = self._make_full_commit_stats(repository, commit)
  File "/Users/user/venv/lib/python3.7/site-packages/githammer/hammer.py", line 170, in _make_full_commit_stats
    for git_object in commit.tree.traverse(visit_once=True):
  File "/Users/user/venv/lib/python3.7/site-packages/git/objects/util.py", line 327, in traverse
    if visit_once and item in visited:
  File "/Users/user/venv/lib/python3.7/site-packages/git/objects/submodule/base.py", line 162, in __hash__
    return hash(self._name)
  File "/Users/user/venv/lib/python3.7/site-packages/gitdb/util.py", line 253, in __getattr__
    self._set_cache_(attr)
  File "/Users/user/venv/lib/python3.7/site-packages/git/objects/submodule/base.py", line 132, in _set_cache_
    raise AttributeError("Cannot retrieve the name of a submodule if it was not set initially")
AttributeError: Cannot retrieve the name of a submodule if it was not set initially
```
Unfortunatelly, I cannot provide the repo itself for reproduction, but I can do some actions for narrowing down the problem if needed.

Maybe https://github.com/gitpython-developers/GitPython/issues/597 and the issue linked at the very bottom can be of help.

**Comment by asharov** [comment]
Hi, thanks for the report and thank you very much for tracking down the source! It is indeed that problem, and it was caused by having a submodule in the initial commit to be processed. I made PR #21 that should fix the problem. Could you check it and verify that it works for you?

**Comment by Azumgi** [comment]
Hey, thank you very much for addressing this so quick! This indeed solved the problem for me :)

**Comment by dkorobtsov** [comment]
Just double checking - this change is not yet available in command line version?
Just updated to version 0.3.0 and got same error (it never happened on this repository before).

**Comment by asharov** [comment]
Thanks for your comment @dkorobtsov ! I meant to release this fix but looks like I forgot. It's now released as 0.3.1 so now the command line version is the same as in the Github repository.

**Comment by dkorobtsov** [comment]
@asharov Thank you! Works just fine!
